components/core/src/clp/DictionaryReader.hpp

```diff
@@ -152,25 +131,36 @@ void DictionaryReader<DictionaryIdType, EntryType>::close() {
         throw OperationFailed(ErrorCode_NotInit, __FILENAME__, __LINE__);
     }
 
-    m_segment_index_decompressor.close();
-    m_segment_index_file_reader.close();
-    m_dictionary_decompressor.close();
-    m_dictionary_file_reader.close();
-
     m_num_segments_read_from_index = 0;
     m_entries.clear();
 
     m_is_open = false;
 }
 
 template <typename DictionaryIdType, typename EntryType>
-void DictionaryReader<DictionaryIdType, EntryType>::read_new_entries() {
-    if (false == m_is_open) {
-        throw OperationFailed(ErrorCode_NotInit, __FILENAME__, __LINE__);
-    }
+void DictionaryReader<DictionaryIdType, EntryType>::read_new_entries(
+        std::string const& dictionary_path,
+        std::string const& segment_index_path
+) {
+    FileReader dictionary_file_reader{dictionary_path};
+    FileReader segment_index_file_reader{segment_index_path};
+
+#if USE_PASSTHROUGH_COMPRESSION
+    streaming_compression::passthrough::Decompressor m_dictionary_decompressor;
+    streaming_compression::passthrough::Decompressor m_segment_index_decompressor;
+#elif USE_ZSTD_COMPRESSION
+    streaming_compression::zstd::Decompressor dictionary_decompressor;
+    streaming_compression::zstd::Decompressor segment_index_decompressor;
+#else
+    static_assert(false, "Unsupported compression mode.");
+#endif
+    constexpr size_t cDecompressorFileReadBufferCapacity = 64 * 1024;  // 64 KB
 
     // Read dictionary header
-    auto num_dictionary_entries = read_dictionary_header(m_dictionary_file_reader);
+    uint64_t num_dictionary_entries{};
```

Comment:
technically we can turn this part into a helper function and put into the dictionary_utils.cpp. But given how simple the code is, I feel it's also ok to leave it as it is.
Let me know if you think it's necessary